pipeline/bootstrap-pipeline.sh

```diff
@@ -0,0 +1,591 @@
+if [[ -z "$TMP" ]]; then
+    TMP=$(mktemp -d) || exit 1
+    trap cleanup EXIT
+fi
+
+# Set ssh key path for potential use with authenticated github
+if [[ ! -z $2 ]]; then
+    ssh_key=$2
+else
+    ssh_key=${HOME}/.ssh/id_rsa
+fi
+set -e
+
+# Figure out if we're using air-gapped machines that should pull images from somewhere other than dockerhub
+extra_params=''
+is_external="false"
+is_internal="false"
+helm_image=
+build_image=
+if [[ "${github}" == "github.com" ]]; then
+    is_external="true"
+    build_image="docker.io/yakinikku/suede_compile"
+    helm_image="docker.io/lachlanevenson/k8s-helm:latest"
+    extra_params="${extra_params} -p build_image=${build_image} -p helm_image=${helm_image}"
+    cp ${repo_root}/pipeline/statefulset.yaml ${TMP}/
+else
+    is_internal="true"
+    build_image="${dockerhub_hostname}/suede_compile:latest"
+    helm_image="${dockerhub_hostname}/k8s-helm"
+    extra_params="${extra_params} -p build_image=${build_image} -p helm_image=${helm_image}"
+    cp ${repo_root}/pipeline/statefulset.yaml ${TMP}/
+    sed -i.bak "s|image: docker.io/yakinikku/suede:latest|image: ${dockerhub_hostname}/suede|g" ${TMP}/statefulset.yaml
+fi
+
+# Figure out if we're running on OpenShift or Kubernetes (kind)
+is_openshift="false"
+is_kubernetes="false"
+client=kubectl
+pipeline_sa=pipeline
+set +e
+kubectl get ns | grep openshift-apiserver
+rc=$?
+if [[ $rc -eq 0 ]]; then
+    is_openshift=true
+    client=oc
+    pipeline_sa=pipeline
+else
+    is_kubernetes=true
+    client=kubectl
+    pipeline_sa=default
+fi
+if [[ ${is_kubernetes} == "true" ]]; then
+    # Assume this is a kind cluster, and install nfs client pvc
+    set -e
+    kubectl apply -f ${repo_root}/pipeline/nfs.yaml
+    helm repo add stable https://charts.helm.sh/stable
+    ip=$(kubectl get svc -n default nfs-service -o jsonpath='{.spec.clusterIP}')
+    helm upgrade --install nfs-provisioner stable/nfs-client-provisioner --values ${repo_root}/pipeline/nfs-values.yaml --set nfs.server=${ip} --namespace nfs-provisioner --create-namespace
+fi
+
+# See if an install namespace will need to be created
+set +e
+rc=1
+if [[ ! -z $1 ]]; then
+    kubectl get ns $1
+    rc=$?
+else
+    kubectl get ns fybrik-system
+    rc=$?
+fi
+
+# Create new project if necessary
+if [[ $rc -ne 0 ]]; then
+    if [[ ${is_openshift} == "true" ]]; then
+        oc new-project ${1:-fybrik-system}
+    else
+        kubectl create ns ${1:-fybrik-system}
+        kubectl config set-context --current --namespace=${1:-fybrik-system}
+    fi
+else
+    if [[ ${is_openshift} == "true" ]]; then
+        oc project ${1:-fybrik-system}
+    else
+        kubectl config set-context --current --namespace=${1:-fybrik-system}
+    fi
+fi
+unique_prefix=$(kubectl config view --minify --output 'jsonpath={..namespace}'; echo)
+
+set +e
+# Be smarter about this - just a quick hack for typical default OpenShift & Kind installs so we can control the default storage class
```

Comment:
is the assumption that `oc` is installed? even if not using OCP
What are the dependencies that we should have?

in the rest of the file there is a lot of use of `oc` not under `${is_openshift} == "true" `

Reply:
No, this should be cleaned up.  Doing that now.

Reply:
Done